resources/views/components/pagination-limit-selector.blade.php

```diff
@@ -0,0 +1,11 @@
+<div>
+    <!-- This component includes limits per pages -->
+
+    <div class="page-actions__label" title="{{ trans('documents.filtering.items_per_page_hint') }}">
+
+        <a href="{{ route($routeName,array_merge($pageParams,['n' => 12])) }}" class="button @if(auth()->user()->optionItemsPerPage() == 12) button--selected @endif">12</a>
```

Comment:
Instead of reading `optionItemsPerPage()` three times you could read it once in the component class and then expose it via property so it will only execute one query against the database.